tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/bs_bt_utils.c

```diff
@@ -0,0 +1,220 @@
+void test_tick(bs_time_t HW_device_time)
+{
+	bs_trace_debug_time(0, "Simulation ends now.\n");
+	if (bst_result != Passed) {
+		bst_result = Failed;
+		bs_trace_error("Test did not pass before simulation ended.\n");
+	}
+}
+
+void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(TEST_TIMEOUT_SIMULATED);
+	bst_result = In_progress;
+}
+
+DEFINE_FLAG(flag_is_connected);
+
+void wait_connected(void)
+{
+	UNSET_FLAG(flag_is_connected);
+	WAIT_FOR_FLAG(flag_is_connected);
+	printk("connected\n");
+}
+
+void wait_disconnected(void)
+{
+	SET_FLAG(flag_is_connected);
+	WAIT_FOR_FLAG_UNSET(flag_is_connected);
+	printk("disconnected\n");
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	UNSET_FLAG(flag_is_connected);
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	if (err != 0) {
+		return;
+	}
+
+	SET_FLAG(flag_is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+void bs_bt_utils_setup(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	ASSERT(!err, "bt_enable failed.\n");
+}
+
+static bt_addr_le_t last_scanned_addr;
+
+static void scan_connect_to_first_result_device_found(const bt_addr_le_t *addr, int8_t rssi,
+						      uint8_t type, struct net_buf_simple *ad)
+{
+	struct bt_conn *conn;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	/* We're only interested in connectable events */
+	if (type != BT_HCI_ADV_IND && type != BT_HCI_ADV_DIRECT_IND) {
+		FAIL("Unexpected advertisement type.");
+	}
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+	printk("Got scan result, connecting.. dst %s, RSSI %d\n", addr_str, rssi);
+
+	err = bt_le_scan_stop();
+	ASSERT(!err, "Err bt_le_scan_stop %d", err);
+
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, BT_LE_CONN_PARAM_DEFAULT, &conn);
+	ASSERT(!err, "Err bt_conn_le_create %d", err);
+
+	/* Save address for later comparison */
+	memcpy(&last_scanned_addr, addr, sizeof(last_scanned_addr));
+}
+
+void scan_connect_to_first_result(void)
+{
+	int err;
+
+	printk("start scanner\n");
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, scan_connect_to_first_result_device_found);
+	ASSERT(!err, "Err bt_le_scan_start %d", err);
+}
+
+static void scan_expect_same_address_device_found(const bt_addr_le_t *addr, int8_t rssi,
+						  uint8_t type, struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	char expected_addr_str[BT_ADDR_LE_STR_LEN];
+
+	/* We're only interested in connectable events */
+	if (type != BT_HCI_ADV_IND && type != BT_HCI_ADV_DIRECT_IND) {
+		FAIL("Unexpected advertisement type.");
+	}
+
+	if (!bt_addr_le_eq(&last_scanned_addr, addr)) {
+		bt_addr_le_to_str(&last_scanned_addr, expected_addr_str, sizeof(expected_addr_str));
+		bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+		FAIL("Expected advertiser with addr %s, got %s\n", expected_addr_str, addr_str);
+	}
+
+	PASS("Advertiser used correct address on resume\n");
+}
+
+void scan_expect_same_address(void)
```

Comment:
This seems to only be used by the central/tester. Could be a static function in `tester.c` if not used for anything else

Reply:
The same seems to apply for multiple other functions. 

Can you please explain me why these have been moved to a common/util file, if they are not common nor used by anything but a single test/file?

Reply:
They haven't been moved there, they have always been there.
They use state from the other common fns, specifically `flag_is_connected` and `last_scanned_addr`.

Reply:
I could remove anything not bsim boilerplate from that file, but we'd have a bunch of duplication. The other issue is this 
```c
BT_CONN_CB_DEFINE(conn_callbacks) = {
	.connected = connected,
	.disconnected = disconnected,
};
```
Since it's the same exe being built for both testers & dut, the callbacks would be called twice on each device. A bunch of bsim tests have this problem actually.

Reply:
Indeed. `BT_CONN_CB_DEFINE` may not be the right way for BSIM tests :)

Reply:
> They haven't been moved there, they have always been there. They use state from the other common fns, specifically `flag_is_connected` and `last_scanned_addr`.

It's fine for now anyways :)

Reply:
> Indeed. `BT_CONN_CB_DEFINE` may not be the right way for BSIM tests :)

yeah it might be better to register at runtime in those cases. We are thinking of making a proper bsim test framework, with a template that can be re-used.